core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -52,8 +60,28 @@ private NetworkInfo getNetworkInfo() {
      * @return boolean true if a Wi-Fi network connection is available.
      */
     public boolean isConnectedToWifi() {
-        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
-        return networkInfo != null && networkInfo.isConnected();
+        if (androidVersion.isMarshmallowOrHigher()) {
```

Comment:
Ah @tasomaniac, I considered this 😄  I would ideally like to have a component that decides which one to use and executes it but not sure if overkill for this PR. I could certainly add a ticket for future reference to make the library a bit cleaner when considering workarounds.